lib/membrane_rtmp_plugin/rtmp/source/message_parser.ex

```diff
@@ -175,19 +180,23 @@ defmodule Membrane.RTMP.MessageParser do
   # in this case the message gets divided into
   # a sequence of smaller packets separated by the a header type 3 byte
   # (the first 2 bits has to be 0b11)
-  defp combine_body_chunks(body, chunk_size) do
+  defp combine_body_chunks(body, chunk_size, header) do
     if byte_size(body) <= chunk_size do
       body
     else
-      do_combine_body_chunks(body, chunk_size, [])
+      do_combine_body_chunks(body, chunk_size, [], header)
     end
   end
 
-  defp do_combine_body_chunks(body, chunk_size, acc) do
+  defp do_combine_body_chunks(body, chunk_size, acc, header) do
     case body do
+      <<body::binary-size(chunk_size), 0b11::2, _chunk_stream_id::6, timestamp::32, rest::binary>>
+      when header.extended_timestamp? and timestamp == header.timestamp ->
```

Comment:
I am not sure how to do this, because the header.timestamp is an integer and not a binary. If matching a short integer it may parse less than 32 bits from the binary and will never match 🤔

I also think its nicer, otherwise I have to extract the timestamp from the header into an own variable.